### PR TITLE
Optimize crossbow processes

### PR DESCRIPTION
**Pull request by Heezay**
1. Written docs about using `--gradle` flag and how to build crossbow Gradle projects. 
2. Created tests directory in crossbundle/cli. Written code to test `execute_apk`, `execute_aab` and `build_gradle` functions directly. 
3. Was deleted condition block in `aapt_add_lib` function.
4. Fixed other insignificant issues.

### CODE REVIEW

**Review by enfipy — CHANGES_REQUESTED** [review]

File: crossbundle/cli/Cargo.toml [hunk 1]
```diff
@@ -39,6 +39,7 @@ dunce = "1.0"
 ureq = "2.4.0"
 cargo = "0.63.1"
 cargo-util = "0.2.0"
+tempfile = "3.2"
```
[the remark this answers is not in the document]
Comment (Heezay) [reply crossbundle/cli/Cargo.toml]: Done.

File: crossbundle/cli/src/commands/build/android.rs [hunk 2]
```diff
@@ -329,11 +331,13 @@ impl AndroidBuildCommand {
         let mut aapt2_link =
             sdk.aapt2()?
                 .link_compiled_res(compiled_res, &apk_path, &manifest_path);
-        aapt2_link
-            .android_jar(sdk.android_jar(target_sdk_version)?)
-            .assets(context.android_assets().unwrap())
-            .proto_format(true)
-            .auto_add_overlay(true);
+        aapt2_link.android_jar(sdk.android_jar(target_sdk_version)?);
+        if let Some(assets) = context.android_assets() {
```
Comment (enfipy) [inline crossbundle/cli/src/commands/build/android.rs:335]: This does return nothing from the `if` scope and why do we need `else`? Perhaps replace with:
```rust
if ... {
    aapt2_link.assets(assets)
} else {
    aapt2_link
}
```

File: crossbundle/cli/src/commands/mod.rs [hunk 3]
```diff
@@ -5,6 +5,7 @@ pub mod new;
 pub mod run;
 
 use crate::error::Result;
+pub use build::*;
```
Comment (enfipy) [inline crossbundle/cli/src/commands/mod.rs:8]: Bad practice. Better replace with `pub mod`.

File: crossbundle/cli/src/lib.rs [hunk 4]
```diff
@@ -7,7 +7,7 @@ pub mod error;
 
 use clap::Parser;
 use colored::Colorize;
-use commands::*;
+pub use commands::{android, BuildContext, Commands, SharedBuildCommand};
```
Comment (enfipy) [inline crossbundle/cli/src/lib.rs:10]: `pub use commands::*`?

File: crossbundle/cli/tests/build_gradle.rs [hunk 5]
```diff
@@ -0,0 +1,54 @@
+use cli::{android::AndroidBuildCommand, BuildContext, SharedBuildCommand};
+use crossbundle_tools::{
+    commands::gen_minimal_project,
+    types::AndroidTarget,
+    utils::{Config, Shell},
+};
+
+#[test]
+/// Use macroquad minimal project in a temporary directory to test gradle project generation. It is working likewise the command below.
```
Comment (enfipy) [inline crossbundle/cli/tests/build_gradle.rs:9]: Make this doc comment in two lines, the current line is too big.

File: crossbundle/cli/tests/execute_aab.rs [hunk 6]
```diff
@@ -0,0 +1,55 @@
+use cli::{android::AndroidBuildCommand, BuildContext, SharedBuildCommand};
+use crossbundle_tools::{
+    commands::gen_minimal_project,
+    types::AndroidTarget,
+    utils::{Config, Shell},
+};
+
+#[test]
+/// Use bevy minimal project in a temporary directory to test AAB generation. It is working likewise the command below.
```
Comment (enfipy) [inline crossbundle/cli/tests/execute_aab.rs:9]: Same.

File: crossbundle/tools/src/commands/common/gen_minimal_project/consts.rs [hunk 7]
```diff
@@ -1,27 +1,37 @@
 pub const BEVY_CARGO_TOML_VALUE: &str = r#"
 [package]
 name = "example"
+version = "0.1.3"
```
Comment (enfipy) [inline crossbundle/tools/src/commands/common/gen_minimal_project/consts.rs:4]: Version in examples shouldn't follow the crossbow version. Replace with `0.1.0`.

File: crossbundle/tools/src/commands/common/gen_minimal_project/consts.rs [hunk 8]
```diff
@@ -1,27 +1,37 @@
...
 authors = ["DodoRare Team <support@dodorare.com>"]
 edition = "2021"
-version = "0.1.0"
 
 [dependencies]
 crossbow = { git = "https://github.com/dodorare/crossbow" }
+
+[package.metadata]
+app_name = "Bevy Example"
```
Comment (enfipy) [inline crossbundle/tools/src/commands/common/gen_minimal_project/consts.rs:12]: "Example"

**Review by enfipy — CHANGES_REQUESTED** [review]

File: crossbundle/cli/src/commands/mod.rs (hunk 3 above)
Comment (Heezay) [reply crossbundle/cli/src/commands/mod.rs:8]: We have `pub mod build` yet. If delete `pub use build::*;` I will not import needed functions `pub use commands::{android, BuildContext, Commands, SharedBuildCommand};`

**Review by enfipy — APPROVED** [review]

**Pull request merged**